### DISCUSSION TITLE
Readonly access to csi secret store provider containers

### DISCUSSION THREAD

**Discussion by shafanaS** [Azure/secrets-store-csi-driver-provider-azure#724] [Q&A]
Does csi secret store provider containers require write access to the file system?
If read only access is sufficient, how can this be enabled through helm chart during deployment.
```
securityContext:
      readOnlyRootFilesystem: true
```

**Comment by nilekhc** [comment]
Hello @shafanaS, Provider does not need write access to file system as secrets are being written by driver. And since it's not required, we have by default set [readOnlyRootFilesystem: true](https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/7e3bd77ff3e1a0447d2bb7e68744b6a619cd8be2/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer.yaml#L69) and won't let this override through helm.

**Reply by shafanaS** [reply]
Hi @nilekhc. Thank you for the answer. 
I have used below chart to deploy csi secret store driver in my AKS.
```
helm repo add csi-secrets-store-provider-azure https://raw.githubusercontent.com/Azure/secrets-store-csi-driver-provider-azure/master/charts
helm upgrade --install csi-secrets-store-provider-azure csi-secrets-store-provider-azure/csi-secrets-store-provider-azure --namespace csi-secret-store-driver --version 0.0.16
```

 According to AKS policy "`Kubernetes cluster containers should run with a read only root file system`" csi secret store containers are listed as non compliant. When I checked the deployment daemonset for the below containers `readOnlyRootFilesystem: true` was not set.

- provider-azure-installer
- node-driver-registrar
- secrets-store
- liveness-probe
- node-driver-registrar


How can I enable `readOnlyRootFilesystem: true ` flag to these containers using helm?

**Reply by nilekhc** [reply]
Hi @shafanaS, I see you are setting version as `0.0.16` in your steps. That is very old version. We now have stable `v1.0.0`.

Could you try these [install step](https://azure.github.io/secrets-store-csi-driver-provider-azure/getting-started/installation/)?
```shell
helm repo add csi-secrets-store-provider-azure https://raw.githubusercontent.com/Azure/secrets-store-csi-driver-provider-azure/master/charts
helm install csi csi-secrets-store-provider-azure/csi-secrets-store-provider-azure --namespace kube-system
```

**Comment by ritazh** [comment]
ICYMI Since you are on AKS, this component is a managed AKS addon, for more information checkout https://docs.microsoft.com/en-us/azure/aks/csi-secrets-store-driver
